api/route/route_test.go

```diff
@@ -98,39 +153,41 @@ func TestCreateRouteWithCreateNewGroup(t *testing.T) {
       "vars":[],
       "upstream":{"type":"roundrobin","nodes":{"127.0.0.1:443":1},
       "timeout":{"connect":6000,"send":6000,"read":6000}},
-      "upstream_header":{},
-      "route_group_id":"",
-      "route_group_name":"route-create-test-create-group"
+      "upstream_header":{}
 }`).Expect(t).
-		Status(http.StatusOK).
-		End()
+	Status(http.StatusOK).
+	End()
+	route, _ := getRouteByName("api-test")
+	// publish route
+	handler.Put(uriPrefix + "/routes/" + route.ID.String() + "/publish").Expect(t).Status(http.StatusOK).End()
 }
 
-func TestCreateRouteWithDuplicateGroupName(t *testing.T) {
-	// create route with duplicate route group name
+func TestOfflineRoute(t *testing.T) {
+	// create route
 	handler.Post(uriPrefix+"/routes").
 		Header("Authorization", token).
 		JSON(`{
-      "name":"api-test",
-      "desc":"api-test",
+      "name":"api-test-published",
+      "desc":"",
       "priority":0,
       "protocols":["http"],
-      "hosts":["test.com"],
+      "hosts":["test1.com"],
       "paths":["/*"],
       "methods":["GET","HEAD","POST","PUT","DELETE","OPTIONS","PATCH"],
-      "status":false,
+      "status":true,
```

Comment:
why is true for offline ?

Reply:
create a route online first, then we could  do `offline` :)